model-optimizer/extensions/ops/ExtractImagePatches.py

```diff
@@ -43,7 +43,9 @@ def infer(node: Node):
 
         assert len(input_shape) == 4, 'ExtractImagePatches operation supports only 4D tensors'
 
-        layout = node.graph.graph['layout']
+        # the operation ExtractImagePatches should always calculate output shape in NHWC layout (because
+        # in OV the operation produces the tensor in NHWC layout)
+        layout = 'NHWC'
```

Comment:
Do not use this code and methods like `shape_for_layout`. Instead, just hardcode dimensions indexes corresponding to layout in NHWC order.

Reply:
Done.